File: Sample/ApplicationSample/sample_cs/ImagePackageUI/ImagePackageUI_Component.cs

class ImagePackageUI_Component_AlphaAnimationComponent : asd.Object2DComponent
{
	int time = 0;
	protected override void OnUpdate()
	{
		// アルファ値をsinカーブでアニメーションするようにする。
		var alpha = (int)((float)(System.Math.Sin(time / 20.0f) + 1.0f) / 2.0f * 255);
		var owner = (asd.TextureObject2D)Owner;
		owner.Color = new asd.Color(255, 255, 255, alpha);
		time++;
	}
}

/// <summary>
/// ImagePackageを用いてUIを配置して、特定の名前の画像をアニメーションさせるサンプル。
/// </summary>
class ImagePackageUI_Component : ISample
{
	public string Description
	{
		get { return "ImagePackageを用いてUIを配置し、その際に特定の名前の画像を\nアニメーションさせるサンプル。"; }
	}

	public string Title
	{
		get { return "ImagePackageとアニメーション"; }
	}

	public string ClassName
	{
		get { return "ImagePackageUI_Component"; }
	}

	public void Run()
	{
		// Altseedを初期化する。
		asd.Engine.Initialize("ImagePackageUI_Component", 640, 480, new asd.EngineOption());

		// イメージパッケージを読み込む
		asd.ImagePackage imagePackage = asd.Engine.Graphics.CreateImagePackage("Data/ImagePackage/UI.aip");

		for (int i = 0; i < imagePackage.ImageCount; i++)
		{
			// テクスチャを取り出す
			asd.Texture2D texture = imagePackage.GetImage(i);
			asd.RectI area = imagePackage.GetImageArea(i);

			// テクスチャをオブジェクトとして追加する
			asd.TextureObject2D textureObject2D = new asd.TextureObject2D();
			textureObject2D.Texture = texture;
			textureObject2D.Position = new asd.Vector2DF(area.X, area.Y);
			asd.Engine.AddObject2D(textureObject2D);

			// Background_Lightという名称の画像にコンポーネントを適用する。
			if (imagePackage.GetImageName(i) == "Background_Light")
			{
				textureObject2D.AlphaBlend = asd.AlphaBlendMode.Add;
				textureObject2D.AddComponent(new ImagePackageUI_Component_AlphaAnimationComponent(), "AlphaAnimation");
			}
		}

		// Altseedのウインドウが閉じられていないか確認する。
		while (asd.Engine.DoEvents())
		{
			// Altseedを更新する。
			asd.Engine.Update();
			Recorder.CaptureScreen("ImagePackageUI_Component", 5, 16, 0.25f, 0.5f);
			Recorder.TakeScreenShot("ImagePackageUI_Component", 30);
		}

		// Altseedの終了処理をする。
		asd.Engine.Terminate();
	}
}

File: Sample/ApplicationSample/sample_cpp/ImagePackageUI/ImagePackageUI_Component.cpp

#include <Altseed.h>

// ImagePackageを用いてUIを配置して、特定の名前の画像をアニメーションさせるサンプル。

class AlphaAnimationComponent 
	: public asd::Object2DComponent
{
	int time = 0;
	void OnUpdate() override
	{
		// アルファ値をsinカーブでアニメーションするようにする。
		auto alpha = (uint8_t) ((float) (sin(time / 20.0f) + 1.0f) / 2.0f * 255);
		auto owner = (asd::TextureObject2D*)GetOwner();
		owner->SetColor(asd::Color(255, 255, 255, alpha));
		time++;
	}
};

void ImagePackageUI_Component()
{
	// Altseedを初期化する。
	asd::Engine::Initialize(asd::ToAString("ImagePackageUI_Component").c_str(), 640, 480, asd::EngineOption());

	// シーンを生成する
	auto scene = std::make_shared<asd::Scene>();

	// レイヤーを生成する
	auto layer = std::make_shared<asd::Layer2D>();

	// シーンにレイヤーを追加する
	scene->AddLayer(layer);

	// シーンを切り替える
	asd::Engine::ChangeScene(scene);

	// イメージパッケージを読み込む
	auto imagePackage = asd::Engine::GetGraphics()->CreateImagePackage(asd::ToAString("Data/ImagePackage/UI.aip").c_str());

	for (int i = 0; i < imagePackage->GetImageCount(); i++)
	{
		// テクスチャを取り出す
		auto texture = imagePackage->GetImage(i);
		auto area = imagePackage->GetImageArea(i);

		// テクスチャをオブジェクトとして追加する
		auto textureObject2D = std::make_shared<asd::TextureObject2D>();
		textureObject2D->SetTexture(texture);
		textureObject2D->SetPosition(asd::Vector2DF(area.X, area.Y));
		layer->AddObject(textureObject2D);

		// Background_Lightという名称の画像にコンポーネントを適用する。
		auto name = imagePackage->GetImageName(i);
		if (asd::astring(imagePackage->GetImageName(i)) == asd::ToAString("Background_Light"))
		{
			textureObject2D->SetAlphaBlendMode(asd::AlphaBlendMode::Add);
			textureObject2D->AddComponent(std::make_shared<AlphaAnimationComponent>(), asd::ToAString("AlphaAnimation"));
		}
	}

	// Altseedのウインドウが閉じられていないか確認する。
	while (asd::Engine::DoEvents())
	{
		// Altseedを更新する。
		asd::Engine::Update();
	}

	// Altseedの終了処理をする。
	asd::Engine::Terminate();

	return;
}